### PR TITLE
F401 use BTreeMap instead of FxHashMap

### LINKED ISSUE #11619: Test failure for 0.4.6 on s390x
The test `rules::pyflakes::tests::preview_rules::rule_unusedimport_path_new_f401_28_all_multiple_init_py_expects` is failing on s390x:

```
---- rules::pyflakes::tests::preview_rules::rule_unusedimport_path_new_f401_28_all_multiple_init_py_expects stdout ----
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot file: crates/ruff_linter/src/rules/pyflakes/snapshots/ruff_linter__rules__pyflakes__tests__preview__F401_F401_28__all_multiple____init__.py.snap
Snapshot: preview__F401_F401_28__all_multiple/__init__.py
Source: crates/ruff_linter/src/rules/pyflakes/mod.rs:228
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    8     8 │ 5 5 | from . import unused, renamed as bees  # F401: add to __all__
    9     9 │ 6 6 | 
   10    10 │ 7 7 | 
   11    11 │ 8   |-__all__ = [];
   12       │-  8 |+__all__ = ["bees", "unused"];
         12 │+  8 |+__all__ = ["unused", "bees"];
   13    13 │ 
   14    14 │ __init__.py:5:34: F401 [*] `.renamed` imported but unused; consider removing, adding to `__all__`, or using a redundant alias
   15    15 │   |
   16    16 │ 5 | from . import unused, renamed as bees  # F401: add to __all__
┈┈┈┈┈┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
   22    22 │ 5 5 | from . import unused, renamed as bees  # F401: add to __all__
   23    23 │ 6 6 | 
   24    24 │ 7 7 | 
   25    25 │ 8   |-__all__ = [];
   26       │-  8 |+__all__ = ["bees", "unused"];
         26 │+  8 |+__all__ = ["unused", "bees"];
────────────┴───────────────────────────────────────────────────────────────────
```

(full CI run log here: https://gitlab.alpinelinux.org/WhyNotHugo/aports/-/jobs/1403448)

I think that the order of the items here is non-deterministic somehow and varies depending on CPU architecture.

### PR DESCRIPTION
**Pull request by plredmond**
* Potentially resolves #11619 (nondeterministic hashmap order across different architectures) in F401 by replacing a hashmap with nondeterministic traversal order with an ordered mapping.

I'm not sure how to test this with our CI/CD. I don't have an s390x machine at home. Should I try it in Qemu?

### CODE REVIEW

File: crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs [hunk 1]
```diff
@@ -220,8 +220,8 @@ fn find_dunder_all_exprs<'a>(semantic: &'a SemanticModel) -> Vec<&'a ast::Expr>
 ///
 pub(crate) fn unused_import(checker: &Checker, scope: &Scope, diagnostics: &mut Vec<Diagnostic>) {
     // Collect all unused imports by statement.
-    let mut unused: FxHashMap<(NodeId, Exceptions), Vec<ImportBinding>> = FxHashMap::default();
-    let mut ignored: FxHashMap<(NodeId, Exceptions), Vec<ImportBinding>> = FxHashMap::default();
+    let mut unused: IndexMap<(NodeId, Exceptions), Vec<ImportBinding>> = IndexMap::default();
+    let mut ignored: IndexMap<(NodeId, Exceptions), Vec<ImportBinding>> = IndexMap::default();
```
[the remark this answers is not in the document]
Comment (plredmond) [reply crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs:224]: Could be. I'll make that change so we don't add the dependency.

[the remark this answers is not in the document]
Comment (plredmond) [reply crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs:224]: Ok to add `Ord` to `ruff_python_semantic::binding::Exceptions`?

**Review by charliermarsh — APPROVED** [review]

**Comment by charliermarsh** [comment]
I would just ship it and ask the original filer if it's resolved.

**Comment by plredmond** [comment]
SGTM. I was having "uncompression error" trying to get an alpine s390x iso to start in qemu. Merging now.

**Pull request merged**